ember-client/shared/src/main/scala/org/http4s/ember/client/internal/WebSocketHelpers.scala

```diff
@@ -38,6 +46,42 @@ private[client] object WebSocketHelpers {
   private[internal] val connectionUpgrade = Connection(NonEmptyList.of(upgradeCi))
   private[internal] val upgradeWebSocket = Upgrade(webSocketProtocol)
 
+  def getSocket[F[_]](client: Client[F], request: Request[F])(implicit
+      F: MonadCancel[F, Throwable]
+  ): Resource[F, Option[Socket[F]]] = {
+    val webSocketKey = WebSocketKey.webSocketConnection[F]
+    client
+      .run(request)
+      .evalMap { res =>
+        for {
+          secWebSocketKeyString <- request.headers
+            .get[`Sec-WebSocket-Key`]
+            .liftTo[F](new RuntimeException("Not found Sec-WebSocket-Key string"))
+            .map(_.hashString)
+          isValid <- validateServerHandshake(res, secWebSocketKeyString)
+        } yield isValid.toOption *> res.attributes.lookup(webSocketKey)
```

Comment:
- It could be good practice to extract the exception as a `val MissingSecWebsocketKey` private to the object.
- Having a `liftTo` (which defaults to error) followed by a `flatMap` can be a bit too implicit. Since that is at an `Option`, perhaps the following suggestion would help? 

```suggestion
        request.headers.get[`Sec-WebSocket-Key`] match {
          case Some(swsv) => 
            validateServerHandshake(res, swsv.hashString)
              .map(if (_) res.attributes.lookup(webSocketKey) else None)
          case None => MissingSecWebsocketKey.raiseError[F]
        }
```

Reply:
Thanks i'll try this